geometry/graham_scan_functions.hpp

```diff
@@ -0,0 +1,193 @@
+* Euclidean space, or equivalently as the set of all convex combinations of 
+* points in the subset. For a bounded subset of the plane, the convex hull may 
+* be visualized as the shape enclosed by a rubber band stretched around the subset.
+*
+* The worst case time complexity of Jarvis’s Algorithm is O(n^2). Using Graham’s 
+* scan algorithm, we can find Convex Hull in O(nLogn) time.
+*
+* ### Implementation
+* 
+* Sort points 
+* We first find the bottom-most point. The idea is to pre-process 
+* points be sorting them with respect to the bottom-most point. Once the points 
+* are sorted, they form a simple closed path.
+* The sorting criteria is to use the orientation to compare angles without actually 
+* computing them (See the compare() function below) because computation of actual 
+* angles would be inefficient since trigonometric functions are not simple to evaluate. 
+* 
+* Accept or Reject Points
+* Once we have the closed path, the next step is to traverse the path and 
+* remove concave points on this path using orientation. The first two points in 
+* sorted array are always part of Convex Hull. For remaining points, we keep track 
+* of recent three points, and find the angle formed by them. Let the three points 
+* be prev(p), curr(c) and next(n). If orientation of these points (considering them 
+* in same order) is not counterclockwise, we discard c, otherwise we keep it. 
+*
+* @author [Lajat Manekar](https://github.com/Lazeeez)
+* 
+*******************************************************************************/
+#include <iostream>     /// for IO operations
+#include <stack>        /// for std::stack
+#include <vector>       /// for std::vector
+#include <algorithm>    /// for std::swap
+#include <cstdlib>      /// for mathematics and datatype conversion
+
+/******************************************************************************
+ * @namespace geometry::grahamscan
+ * @brief geometric algorithms
+ *******************************************************************************/
+namespace geometry::grahamscan {
+
+    /******************************************************************************
+    * @struct Point
+    * @brief for X and Y co-ordinates of the co-ordinate.
+    *******************************************************************************/
+    struct Point {
+        int x, y;
+    };
+
+    // A global point needed for sorting points with reference
+    // to  the first point Used in compare function of qsort()
+    Point p0;
+
+    /******************************************************************************
+    * @brief A utility function to find next to top in a stack.
+    * @param S Stack to be used for the process.
+    * @returns @param Point Co-ordinates of the Point <int, int>
+    *******************************************************************************/
+    Point nextToTop(std::stack<Point> &S) {
```

Comment:
Let's make this a `const` as it's not modified but only being accessed. 😄
```suggestion
    Point nextToTop(const std::stack<Point> &S) {
```

Reply:
Oh right!